File: Task-03/Ruby/Sub-Task4.rb
def generate_diamond(n)
  diamond = ""

  (0...n).each do |i|
    diamond += " " * (n - i - 1) + "* " * (i + 1) + "\n"
  end
  
  (n - 2).downto(0) do |i|
    diamond += " " * (n - i - 1) + "* " * (i + 1) + "\n"
  end

  diamond
end

begin
  
  input_file = 'input.txt'
  output_file = 'output.txt'

  n = File.read(input_file).to_i

  if n > 0
    diamond_pattern = generate_diamond(n)

    File.open(output_file, 'w') do |file|
      file.write(diamond_pattern)
    end

  end
end

File: Task-03/Elixir/Sub-Task4.exs
defmodule DiamondPattern do

  def main do
    n = File.read!("input.txt") |> String.trim() |> String.to_integer()
    diamond_pattern = generate_diamond(n)
    File.write!("output.txt", diamond_pattern)
    IO.puts "Diamond pattern generated and saved to output.txt"
  end

  defp generate_diamond(n) do
    top_half = Enum.map(1..n, fn i -> print_line(i, n) end)
    bottom_half = Enum.map((n-1)..1, fn i -> print_line(i, n) end)
    Enum.join(top_half ++ bottom_half, "\n")
  end

  defp print_line(i, n) do
    spaces = n - i
    stars = 2 * i - 1
    String.duplicate(" ", spaces) <> String.duplicate("*", stars)
  end
end

DiamondPattern.main()
